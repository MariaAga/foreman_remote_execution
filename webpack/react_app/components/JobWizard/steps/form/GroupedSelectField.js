import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  SelectOption,
  Select,
  SelectGroup,
  SelectVariant,
  FormGroup,
} from '@patternfly/react-core';
/* typeahead issue fixed in 2020.14 */
export const GroupedSelectField = ({
  label,
  fieldId,
  groups,
  selected,
  setSelected,
}) => {
  const [isOpen, setIsOpen] = useState(false);
  const onSelect = selection => {
    setIsOpen(false);
    setSelected(selection);
  };

  const onClear = () => {
    onSelect(null);
  };
  const options = groups.map((group, groupIndex) => (
    <SelectGroup key={groupIndex} label={group.groupLabel}>
      {group.options.map((option, optionIndex) => (
        <SelectOption
          key={optionIndex}
          value={option.label}
          onClick={() => onSelect(option.value)}
        />
      ))}
    </SelectGroup>
  ));

  return (
    <FormGroup label={label} fieldId={fieldId}>
      <Select
        isGrouped
        variant={SelectVariant.typeahead}
        onToggle={setIsOpen}
        isOpen={isOpen}
        onSelect={() => null}
        selections={selected}
        className="without_select2"
        onClear={onClear}
      >
        {options}
      </Select>
    </FormGroup>
  );
};

GroupedSelectField.propTypes = {
  label: PropTypes.string.isRequired,
  fieldId: PropTypes.string.isRequired,
  groups: PropTypes.array,
  selected: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  setSelected: PropTypes.func.isRequired,
};

GroupedSelectField.defaultProps = {
  groups: [],
  selected: null,
};
