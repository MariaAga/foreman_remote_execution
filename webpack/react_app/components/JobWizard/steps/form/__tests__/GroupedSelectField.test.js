import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { GroupedSelectField } from '../GroupedSelectField';

const fixtures = {
  'renders with props': {
    label: 'grouped select',
    fieldId: 'field-id',
    groups: [
      {
        groupLabel: 'Ansible',
        options: [
          {
            label: 'Ansible Roles - Ansible Default',
            value: 168,
          },
          {
            label: 'Ansible Roles - Install from git',
            value: 170,
          },
        ],
      },
    ],
    selected: 170,
    setSelected: jest.fn(),
  },
};

describe('GroupedSelectField', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(GroupedSelectField, fixtures));
});
