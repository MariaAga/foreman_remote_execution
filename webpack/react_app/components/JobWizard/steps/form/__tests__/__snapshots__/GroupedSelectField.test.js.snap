// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupedSelectField rendering renders with props 1`] = `
<FormGroup
  fieldId="field-id"
  label="grouped select"
>
  <Select
    aria-label=""
    aria-labelledby=""
    className="without_select2"
    clearSelectionsAriaLabel="Clear all"
    createText="Create"
    customBadgeText={null}
    customContent={null}
    direction="down"
    favorites={Array []}
    favoritesLabel="Favorites"
    hasInlineFilter={false}
    inlineFilterPlaceholderText={null}
    inputIdPrefix=""
    isCreatable={false}
    isDisabled={false}
    isGrouped={true}
    isOpen={false}
    isPlain={false}
    menuAppendTo="inline"
    noResultsFoundText="No results found"
    onClear={[Function]}
    onCreateOption={[Function]}
    onFilter={null}
    onSelect={[Function]}
    onToggle={[Function]}
    ouiaSafe={true}
    placeholderText=""
    removeSelectionAriaLabel="Remove"
    selections={170}
    toggleAriaLabel="Options menu"
    toggleIcon={null}
    toggleId={null}
    typeAheadAriaLabel=""
    variant="typeahead"
    width=""
  >
    <SelectGroup
      key="0"
      label="Ansible"
    >
      <SelectOption
        className=""
        component="button"
        index={0}
        inputId=""
        isChecked={false}
        isDisabled={false}
        isFavorite={null}
        isNoResultsOption={false}
        isPlaceholder={false}
        isSelected={false}
        key="0"
        keyHandler={[Function]}
        onClick={[Function]}
        sendRef={[Function]}
        value="Ansible Roles - Ansible Default"
      />
      <SelectOption
        className=""
        component="button"
        index={0}
        inputId=""
        isChecked={false}
        isDisabled={false}
        isFavorite={null}
        isNoResultsOption={false}
        isPlaceholder={false}
        isSelected={false}
        key="1"
        keyHandler={[Function]}
        onClick={[Function]}
        sendRef={[Function]}
        value="Ansible Roles - Install from git"
      />
    </SelectGroup>
  </Select>
</FormGroup>
`;
